File: txxx.js
const axios = require('axios');
const cheerio = require('cheerio');

module.exports = async function txxx(query) {
  const url = `https://www.txxx.com/search/${encodeURIComponent(query)}/`;
  const results = [];

  try {
    const { data } = await axios.get(url);
    const $ = cheerio.load(data);

    $('div.video-box').each((i, el) => {
      const title = $(el).find('.title a').text().trim();
      const href = $(el).find('a').attr('href');
      const duration = $(el).find('.duration').text().trim();

      if (title && href) {
        results.push({
          title,
          url: 'https://www.txxx.com' + href,
          duration,
          source: "TXXX"
        });
      }
    });
  } catch (err) {
    console.error("txxx error:", err.message);
  }

  return results;
};

File: scrapers/txxx.py
import requests
from bs4 import BeautifulSoup

def scrape_txxx(query):
    url = f"https://www.txxx.com/search/{query}/"
    headers = {
        "User-Agent": "Mozilla/5.0"
    }
    res = requests.get(url, headers=headers)
    soup = BeautifulSoup(res.text, "html.parser")

    results = []
    for video in soup.select(".thumb"):
        a = video.select_one("a")
        title = a["title"]
        link = "https://www.txxx.com" + a["href"]
        preview = a.select_one("img")["data-original"]

        results.append({
            "title": title,
            "url": link,
            "preview": preview,
            "source": "TXXX"
        })

    return results
